charts/hail-search/templates/deployment.yaml

```diff
@@ -11,8 +11,9 @@ spec:
       {{- include "hail-search.selectorLabels" . | nindent 6 }}
   template:
     metadata:
-      {{- with .Values.podAnnotations }}
       annotations:
+        checksum/datasetVersions: {{ print $.Values.datasetVersions | sha256sum }}
```

Comment:
I believe this forces creation of a new `Deployment` upon a change in any of the `datasetVersions`.  It's following the pattern we have elsewhere.

Reply:
yup, this should do it, and this is the recommended way to signal to a deployment that some external dependency has changed, which otherwise wouldn't restart pods: https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments